src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1776,46 +1776,41 @@
     </Field>
     <Field Name="Designation" Type="VariableDesignationSyntax"/>
   </Node>
-  <Node Name="DeconstructionPatternSyntax" Base="PatternSyntax">
-    <Kind Name="DeconstructionPattern" />
+  <Node Name="RecursivePatternSyntax" Base="PatternSyntax">
```

Comment:
Not sure how if this is important but not all recursive patterns could be defined under this type in the future e.g. list/or patterns. I think ObjectPattern is more accurate here.